pkg/client/daemon/outbound.go

```diff
@@ -187,9 +207,8 @@ func (o *outbound) resolveInCluster(c context.Context, qType uint16, query strin
 		}
 	}
 
-	// Give the cluster lookup a reasonable timeout
-	tos := client.GetConfig(c).Timeouts
-	c, cancel := tos.TimeoutContext(c, client.TimeoutTrafficManagerAPI)
+	// Give the cluster lookup a reasonable timeout.
+	c, cancel := context.WithTimeout(c, 4*time.Second)
```

Comment:
I think we should make this timeout configurable, add a TODO comment saying "we * should * make this configurable at some point", or provide a reason why this should be hardcoded to 4 seconds.

Reply:
That's an oversight. This timeout should definitely be added.